### DISCUSSION TITLE
Troubles with plugin permissions

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5352] [Q&A Core] [answered]
Hi,
I have an admin with permission to access one plugin, but nothing else.  

If I use old code:
```
if (!getperms('P'))
{
	e107::redirect('admin');
	exit;
}
```
They have access as they should have.

If I use this:
```
if (!e107::getUser()->checkAdminPerms('P'))
{
	e107::redirect('admin');
	exit;
}
```
They don't have access. Plugin is visible but it is returning to dashboard page. 

@Deltik What is correct replacement for outdated permission check?  Thanks

Next question:
If my admin menu perms look like this:
```
		'main/prefs' 		=> array('caption' => LAN_PREFS, 'perm' => 'P', 'url' => 'admin_config.php'),
		'main/list' 		=> array('caption' => LAN_ADMIN_RACE, 'perm' => '0', 'url' => 'admin_config.php'),
		'main/create' 		=> array('caption' => LAN_ADMIN_ADD_RACE, 'perm' => '0', 'url' => 'admin_config.php'),
```

and  "P"  permission is needed for file admin_config.php 
then if an admin is familiar with e107 and guess URLs for creating or listing actions, they will have access to them.    
When I asked about this years ago, it was recommended to check permission again in init() - see faqs example 

So I used
```
	function init()
	{ 
		$this->access = array('main/list'   => '0');
	}
```

And URL is still available for plugin admin.  Any ideas?    

For now, I will split code to more files, to be able to control this with if (!getperms('0')) 

Thank

**Comment by Deltik** [comment, marked as answer]
The replacement for `getperms('P')` is `e_user_model::checkPluginAdminPerms()`.  The method is documented [at `getperms()`](https://github.com/e107inc/e107/commit/dd36fbd51db1cd80d775a82a9b53287ad0aa7223#diff-78d2eb1af2672cd1b71e61042086939c194480b4bf9d9733b68f0e800a8a134bR1335) and [at the new method](https://github.com/e107inc/e107/commit/dd36fbd51db1cd80d775a82a9b53287ad0aa7223#diff-5d42692f7ba27af5aacf29b292c11e2dc15a3ced9ba8d43ad049623644cd7388R670).

The new method doesn't assume the plugin path, so it's not affected by the file path where the permission check is used.  This should also address your second concern.

**Reply by Jimmi08** [reply]
I just noticed that in this commit 

https://github.com/e107inc/e107/commit/7d94bca7d6c95f1853902c9cac31d49e971bdbc1

it was replaced with getperms() again

**Reply by Deltik** [reply]
Thanks for pointing that out, @Jimmi08.  I brought up the concern with @CaMer0n in https://github.com/e107inc/e107/issues/5476#issuecomment-3023999163.
